### ISSUE TITLE
Comparison / collaboration

### ISSUE THREAD

**Issue by Raynos** [josephsavona/valuable#1]
I've been working on the exact same idea (with minor interface differences) with observ.

`observ` is the model layer in mercury ( https://github.com/Raynos/mercury#state-modules-the-model-layer ).

It consists of 
- [`observ`](https://github.com/Raynos/observ) Same as `Valuable`
- [`observ-struct`](http://ghub.io/observ-struct) Same as `Struct`
- [`observ-varhash`](http://ghub.io/observ-varhash) Same as `Map`
- [`observ-array`](http://ghub.io/observ-array) Same as `List`

The key value proposition of valuable seems to be arbitrary nested structures and change bubbling. This is the same key value proposition of `observ-varhash` and `observ-array`.

The technique of arbitrary nested data structures and bubbling of changes is called lenses ( http://adit.io/posts/2013-07-22-lenses-in-pictures.html )

This is painfully non-trivial to implement correctly, since it's effectively replication.

Take your simple README example:

``` js
var value = Valuable({
  items: [1,2,3],
  name: 'numbers'
});

var secondV = value.get('items')

value.setValue({ items: [1, 2, 3, 4] })
assert.equal(secondV.val(3), 4)
```

Basically bubbling upwards is half the problem, you also need to bubble downwards otherwise global undo/redo breaks.

Since this is painfully non-trivial and there is a lot of overlap between `observ-*` and `valuable` I would love to collaborate on this problem.
